src/test/java/tsunami/security/scanner/utilities/KubeJavaClientUtilTest.java

```diff
@@ -0,0 +1,111 @@
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.models.V1DeleteOptions;
+import io.kubernetes.client.util.Config;
+import java.io.IOException;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class KubeJavaClientUtilTest {
+
+  @Test
+  public void testKubeJavaClientUtil() throws IOException, ApiException, InterruptedException {
+    String resourceConfig =
+        "apiVersion: v1\n"
+            + "kind: Service\n"
+            + "metadata:\n"
+            + "  name: jupyter\n"
+            + "  labels:\n"
+            + "    app: jupyter\n"
+            + "spec:\n"
+            + "  ports:\n"
+            + "  - port: 80\n"
+            + "    name: http\n"
+            + "    targetPort: 8888\n"
+            + "  selector:\n"
+            + "    app: jupyter\n"
+            + "  type: LoadBalancer\n"
+            + "---\n"
+            + "apiVersion: v1\n"
+            + "kind: Pod\n"
+            + "metadata:\n"
+            + "  name: jupyter\n"
+            + "  labels:\n"
+            + "    app: jupyter\n"
+            + "spec:\n"
+            + "  containers:\n"
+            + "    - name: jupyter\n"
+            + "      image: skippbox/jupyter:0.0.3\n"
+            + "      ports:\n"
+            + "      - containerPort: 8888\n"
+            + "        protocol: TCP\n"
+            + "        name: http\n"
+            + "      volumeMounts:\n"
+            + "        - mountPath: /root\n"
+            + "          name: notebook-volume\n"
+            + "  volumes:\n"
+            + "  - name: notebook-volume\n"
+            + "    gitRepo:\n"
+            + "      repository: \"https://github.com/kubernetes-client/python.git\"\n";
+
+    ApiClient client = Config.defaultClient();
+    Configuration.setDefaultApiClient(client);
+
+    KubeJavaClientUtil.createResources(resourceConfig);
+
+    Thread.sleep(1000);
+
+    List<String> podList = KubeJavaClientUtil.getPods();
+    List<String> serviceList = KubeJavaClientUtil.getServices();
+
+    // Check started services and pods.
+    assertThat(serviceList).contains("jupyter");
+    assertThat(podList).contains("jupyter");
+
+    // Delete all created resources.
+    CoreV1Api coreV1Api = new CoreV1Api();
+
+    coreV1Api.deleteNamespacedService(
+        "jupyter", "default", null, null, null, null, null, new V1DeleteOptions());
+
+    try {
+      coreV1Api.deleteNamespacedPod(
+          "jupyter", "default", null, null, null, null, null, new V1DeleteOptions());
+    } catch (JsonSyntaxException e) {
+      if (e.getCause() instanceof IllegalStateException) {
+        IllegalStateException ise = (IllegalStateException) e.getCause();
+        if (ise.getMessage() == null
+            || !ise.getMessage().contains("Expected a string but was BEGIN_OBJECT")) throw e;
+      } else throw e;
+    }
+
+    Thread.sleep(10000);
```

Comment:
Please avoid calling external services in unit test. It is fine doing so in integration test, but never do similar things in unit test. 

Since the k8s Java client api doesn't provide a fake client, you could mock the `CoreV1Api` using [Mockito](https://github.com/mockito/mockito) for tseting. In this case you need to redesign the class a little bit to receive `CoreV1Api` as a constructor dependency and pass a mocked `CoreV1Api` within unit test.

Reply:
Still in process, same as AppTest.java